src/visitors/Tag.js

```diff
@@ -54,8 +59,19 @@ function getAttributes(
 
       const expr = parseExpression(val === true ? 'true' : val, context);
 
-      if (!mustEscape && (!t.isStringLiteral(expr) || /(\<\>\&)/.test(val))) {
-        throw new Error('Unescaped attributes are not supported in react-pug');
+      if (!mustEscape) {
+        const isStringViaAliases =
+          t.isStringLiteral(expr) && name !== 'className' && name !== 'id';
+
+        const isNotStringOrBoolean =
+          !t.isStringLiteral(expr) && !t.isBooleanLiteral(expr);
```

Comment:
I'm just wondering if this lot can be clarified. It seems like the allowed case is just:

```js
t.isBooleanLiteral(expr) ||
((name === 'className' || name === 'id') && t.isStringLiteral(expr))
```

It seems simpler written like that to me, rather than having multiple references to `isStringLiteral`.  How about we just use:

```js
const isAllowedUnescapedAttribute = (
  t.isBooleanLiteral(expr) ||
  ((name === 'className' || name === 'id') && t.isStringLiteral(expr))
);
if (!isAllowedUnescapedAttribute) {
```

Reply:
@ForbesLindesay I agree that my naming can confuse, but it was created with following reason in mind: all not obvious conditions should be commented. Variables are more convenient than just comments, so I used them.

`isAllowedUnescapedAttribute` sounds not obvious because we don't use such naming through this plugin and pug (I mean it does not respect spirit of `mustEscape`). It's better to keep `shouldBeEscaped`, `allowedToBeEscaped`.

### Let me show how did I achieve my code:

When attribute is marked that we can skip its escaping, we should always show a message about unsupported feature. Two exceptions, where we should not show a message: 

1. if attribute is a string not from aliases (not from `className` and `id`)
2. if attribute is not primitive (in pug it's everything besides string and booleans)

If I convert it to the code:

```js
if (!mustEscape) {
  const isStringNotFromAliases = 
    t.isStringLiteral(expr) && name !== 'className' && name !== 'id';

  const isNotPrimitive =
    !t.isStringLiteral(expr) && !t.isBooleanLiteral(expr);

  if (isStringNotFromAliases || isNotPrimitive) {
    throw context.error(
      'INVALID_EXPRESSION',
      'Unescaped attributes are not supported in react-pug',
    );
  }
}
```

This snippet is what I suggest to leave. But I don't mind to make it much more compact by refactoring javascript, not the logic. If this is the case, then could you suggest any concrete solution that I can use? Maybe you can help me with optimizing the logic?

Reply:
I'd much rather see a whitelist of cases where it is ok to have unescaped attributes:

1. `className` and `id` if the value is a string literal
2. any attribute if the value is a boolean

As this is much easier for me to interpret. When I read your code, I have to mentally figure out which are the allowed cases. Since the allowed cases are so narrow, I never really feel like I need to figure out what the non-allowed cases are.

`allowedToBeEscaped` doesn't make sense because everything *can* be escaped. `mustEscape` is fine, but we already have that with a subtly different meaning. I think something like `escapingNotRequired` or `canSkipEscaping` makes more sense here.

Reply:
@ForbesLindesay you are right that understanding whitelist is much better here. But I still need your help with figuring out the right way. Does following code look better:

```js
if (!mustEscape) {
  const canSkipEscaping =
    (name === 'className' || name === 'id') && t.isStringLiteral(expr);

  if (!canSkipEscaping) {
    throw context.error(
      'INVALID_EXPRESSION',
      'Unescaped attributes are not supported in react-pug',
    );
  }
}
```

I noticed that we should not check is the value boolean or not. We don't allow to write `!=` constructions at all. Could you please take a look at tests in this PR and tell me is everything appropriate there?

Reply:
Yes, although you've dropped the `BooleanLiteral` case there. Is that inentional?